### PR TITLE
[incubator/anchore-engine] Adds incubator chart for Anchore Engine

### PR DESCRIPTION
**Pull request by zhill**
Adds Anchore Engine chart to incubator.

To test the service you can hit the API (as described in the NOTES.txt) or use the anchore-cli to query the system. The cli is available [here](https://github.com/anchore/anchore-cli) or via pip: `pip install anchorecli`. The instructions in NOTES indicates how to configure the cli.

### CODE REVIEW

**Comment by cblecker** [comment]
bump

**Comment by zhill** [comment]
Working on the linter issues in the CircleCI result. Will update with commit to resolve those.

File: incubator/anchore-engine/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ - security
+home: https://anchore.io
+sources:
+ - https://github.com/anchore/anchore-engine
+maintainers:
+ - name: zhill
+   email: zach@anchore.com
+engine: gotpl
+icon: https://anchore.com/wp-content/uploads/2016/08/anchore.png
+appVersion: 0.1.3
```
Comment (viglesiasce) [inline incubator/anchore-engine/Chart.yaml]: nit, but can you move this to just under the version above.

File: incubator/anchore-engine/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+Anchore Engine Helm Chart
+=========================
+
+This chart deploys the Anchore Engine docker container image analysis system. Anchore Engine
+requires a PostgresSQL database (>=9.6) which may be handled by the chart of supplied externally,
```
Comment (viglesiasce) [inline incubator/anchore-engine/README.md]: /of/or/

File: incubator/anchore-engine/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+Here are the steps to configure the anchore-cli (`pip install anchorecli`). Use these same values for direct API access as well.
+
+To configure your anchore-cli run:
+
+    ANCHORE_CLI_USER=admin
+    ANCHORE_CLI_PASS=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.data.adminPassword}" | base64 --decode; echo)
+
+To get the service endpoint:
+
+If using an ingress:
```
Comment (viglesiasce) [inline incubator/anchore-engine/templates/NOTES.txt]: Can you conditionally template this in/out based on whether the ingress is enabled?

File: incubator/anchore-engine/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    ANCHORE_CLI_URL=$(kubectl get ingress --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+
+If not using ingress, but using ClusterIP or NodePort, then from within the cluster you can use:
+    ANCHORE_CLI_URL={{ template "fullname" . }}:{{ .Values.service.ports.api}}/v1/
+
+or to use the IP instead of service dns name:
+
+    ANCHORE_CLI_URL=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.spec.clusterIP}")":{{.Values.service.ports.api}}/v1/"
```
Comment (viglesiasce) [inline incubator/anchore-engine/templates/NOTES.txt]: Can you add a "smoke test" here? How do users know that Anchore is installed properly? What can they do with it once its installed? Maybe also point to the user docs.

**Review by viglesiasce — CHANGES_REQUESTED** [review]

**Comment by zhill** [comment]
I have addressed the review suggestions and also added a commit to clean up some labelling issues in the deployment definitions to ensure things like 'kubectl get pods -l' filters behave as expected.

**Comment by zhill** [comment]
Moved chart from incubator to stable for PR review per discussion with @viglesiasce

**Comment by viglesiasce** [comment]
Worked a treat! Thanks @zhill!

/lgtm

**Review by viglesiasce — APPROVED** [review]

**Comment by zhill** [comment]
Thanks @viglesiasce !

**Pull request merged**